coala_quickstart/generation/Project.py

```diff
@@ -50,6 +52,22 @@ def language_percentage(file_paths):
         if ext in exts:
             for lang in exts[ext]:
                 results[lang] += delta
+        elif not ext:
+            with open(file_path, 'r') as data:
+                hashbang = data.readline()
+                if(re.match(
+                        '/(^#![(a-z)|\/]*[ ][a-z]*)|(^#![(a-z)|\/]*)/',
+                        hashbang)):
+                    language = get_language_from_hashbang(hashbang)
+                    try:
+                        for ext in exts:
+                            for lang in exts[ext]:
+                                if language == lang.lower():
+                                    results[lang] += delta
+                    except KeyError:
```

Comment:
Why would an exception be triggered? `results` is a `defaultdict` - if a key doesn't exist, it's created.'

Plus, `try-except`s are computationally costly. You should prefer a `if key in dict`.